modin/backends/pandas/query_compiler.py

```diff
@@ -201,6 +201,9 @@ def default_to_pandas(self, pandas_op, *args, **kwargs):
         else:
             return result
 
+    def finalize(self):
+        self._modin_frame.finalize()
```

Comment:
What about other backends? (`pyarrow`, `omnisci`) Should we override for these?

Reply:
We should override it for every backend which defers operations, so most likely yes.

Reply:
Do note that for OmniSci it is already overridden (with `raise NotImplementedError()` stub, cc @ienkovich), and `pyarrow` inherits the one from Pandas so it doesn't seem to be necessary for now.